src/cluster/placement/algo/mirrored.go

```diff
@@ -225,6 +237,49 @@ func (a mirroredAlgorithm) ReplaceInstances(
 	return p, nil
 }
 
+func (a mirroredAlgorithm) markInstanceAndItsPeersAvailable(
```

Comment:
Debatable, but maybe worth adding some targeted unittests for this one (in addition to what you've got)?

Reply:
sounds good.